extension/httpfs/httpfs.cpp

```diff
@@ -499,8 +500,25 @@ void HTTPFileHandle::Initialize(FileOpener *opener) {
 			length = 0;
 			return;
 		} else {
-			throw IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" + res->error +
-			                  ")");
+			// HEAD request fail, use Range requet for another try (read only one byte)
+			if ((flags & FileFlags::FILE_FLAGS_READ)) {
+				char dummy_buffer[2];
+				auto range_res = hfs.GetRangeRequest(*this, path, {}, 0, dummy_buffer, 2);
+				if (range_res->code != 206) {
+					throw IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" +
```

Comment:
Our Exception class (and by inheritance, IOException as well) has support for formatted strings, which allows us to use:
```c++
IOException("Unable to connect to URL \"%s\": %d (%s)", path, res->code, res->error);
```
instead of:
```c++
IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" +
					                  res->error + ");
```